src/lib/__test__/diff.test.js

```diff
@@ -0,0 +1,241 @@
+import { diff } from '../diff';
+
+const clone = obj => JSON.parse(JSON.stringify(obj));
+
+describe('diff', () => {
```

Comment:
We probably should test each of the utilities in its own test file, but for the time being, I made one test for everything in case we want to try another approach (see PR description).